flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerNG.java

```diff
@@ -84,7 +85,11 @@
 
 	void handleGlobalFailure(Throwable cause);
 
-	boolean updateTaskExecutionState(TaskExecutionState taskExecutionState);
+	default boolean updateTaskExecutionState(TaskExecutionState taskExecutionState) {
+		return updateTaskExecutionState(TaskExecutionStateWrapper.createFrom(taskExecutionState));
```

Comment:
This seems like a dangerous default, and I wouldn't be surprised if we at some point call this by accident instead of creating a failed `TaskExecutionStateWrapper`

Reply:
You are right. However, this method will be removed soon when removing the legacy scheduling in ExecutionGraph (FLINK-15626). And I hope to not refactor its usages right now to avoid massive conflicts with the ongoing change of FLINK-17760.
I will mark it as deprecated for now to prevent it from being further used?
WDYT?

Reply:
sounds good